dask/array/core.py

```diff
@@ -1404,7 +1404,11 @@ def chunks(self, chunks):
     def __len__(self):
         if not self.chunks:
             raise TypeError("len() of unsized object")
-        return sum(self.chunks[0])
+        try:
+            return int(sum(self.chunks[0]))
+        except (TypeError, ValueError):
+            pass
```

Comment:
It looks like we're catching multiple failure modes here. I wonder if we might be more explicit and do something like 

https://github.com/dask/dask/blob/d3f7cd248fd5060ba882804f89848f669ece93e2/dask/array/core.py#L1739

instead where we check is there are any `np.nan`s in `self.chunks[0]`.

Reply:
I agree this is easier to understand